### ISSUE TITLE
az vm: 'repair' is not in the 'az vm' command group

### ISSUE THREAD

**Issue by vamsivellanki** [Azure/azure-cli-extensions#2025]
When executing this az vm repair command getting error as below
Executed command: az vm repair create -g TestAz -n TestVM
Error: **az vm: 'repair' is not in the 'az vm' command group**. See 'az vm --help'. If the command is from an extension, please make sure the corresponding extension is installed. To learn more about extensions, please visit https://docs.microsoft.com/en-us/cli/azure/azure-cli-extensions-overview

Can you add this command to the latest module/version to defaults?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a0c5cd11-36e0-b233-4f0c-3d432a9925a1
* Version Independent ID: 99b6e064-4f75-9cce-b6e0-0ad1943552b9
* Content: [az vm repair](https://docs.microsoft.com/en-us/cli/azure/ext/vm-repair/vm/repair?view=azure-cli-latest)
* Content Source: [latest/docs-ref-autogen/ext/vm-repair/vm/repair.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/live/latest/docs-ref-autogen/ext/vm-repair/vm/repair.yml)
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by yungezz** [comment]
hi @qwordy could you pls help to have a look?

**Comment by qwordy** [comment]
Run this command.
```
az extension add -n vm-repair
```
Adding it to azure cli main repo depends on multiple factors including the usage data.
@fengzhou-msft Could you share your plan about this issue? A more helpful message? Install it automatically when invoking?

**Comment by vamsivellanki** [comment]
Yes, only after adding this extension we are able to run the az vm repair command.
This is most useful command while troubleshooting issues on VMs when there is need to create VMs and attaching disks
Installing command when invoked will be helpful if this is not added in module by default or adding the command "az extension add -n vm-repair" to the document https://docs.microsoft.com/en-us/cli/azure/ext/vm-repair/vm/repair?view=azure-cli-latest will helpful.

**Comment by vamsivellanki** [comment]
@fengzhou-msft , could you provide update?

**Comment by fengzhou-msft** [comment]
@vamsivellanki we are developing a feature to dynamically install extensions for the command you run: https://github.com/Azure/azure-cli/pull/14478 ETA is to release on August 4th.

**Comment by yungezz** [comment]
hi @vamsivellanki the dynamic extension installation feature is released already. here's the guideline of how to enable it: https://docs.microsoft.com/en-us/cli/azure/azure-cli-extensions-overview?view=azure-cli-latest#install-extensions-automatically. will close the issue for now. Feel free to create new issue if any other ask. thanks.
